django_restql/mixins.py

```diff
@@ -552,4 +553,142 @@ def update(self, instance, validated_data):
             fields["many_to"]["one_related"]
         )
 
-        return super().update(instance, validated_data)
\ No newline at end of file
+        return super().update(instance, validated_data)
+
+
+class EagerLoadingMixin(object):
+    @property
+    def parsed_query(self):
+        """
+        Gets parsed query for use in eager loading. Defaults to the serializer parsed query assuming
+        using django-restql DynamicsFieldMixin.
+        """
+        if hasattr(self, "get_serializer_class"):
+            serializer_class = self.get_serializer_class()
+
+            if hasattr(serializer_class, "query_param_name"):
+                return serializer_class.get_parsed_query_from_req(self.request)
+
+    def get_queryset(self):
+        """
+        Override for DRF's get_queryset on the view. If get_queryset is not present, we don't try to
+        run this. Instead, this can still be used by manually calling self.get_eager_queryset and
+        passing in the queryset desired.
+        """
+        if hasattr(super(), "get_queryset"):
+            queryset = super().get_queryset()
+            queryset = self.get_eager_queryset(queryset)
+            return queryset
+
+    def get_prefetch_related_mapping(self):
+        if hasattr(self, "prefetch_related"):
+            return self.prefetch_related
+        return {}
+
+    def get_select_related_mapping(self):
+        if hasattr(self, "select_related"):
+            return self.select_related
+        return {}
+
+    def get_parsed_dict(self, data=None):
+        """
+        Returns the parsed query as a dict.
+        """
+        keys = {}
+        if data is None:
+            data = self.parsed_query
+
+        if data is not None:
+            for item in data:
+                if isinstance(item, str):
+                    keys[item] = None
+                elif isinstance(item, dict):
+                    for key, nested_items in item.items():
+                        key_base = key
+                        nested_keys = self.get_parsed_dict(nested_items)
+                        keys[key_base] = nested_keys
```

Comment:
No need to check data twice, it would be great to use if-else.

Reply:
We do need to check because we reassign from `self.parsed_query` which might return `None`.

Reply:
Oh I see.